internal/command/command.go

```diff
@@ -22,6 +22,69 @@ type Command interface {
 	Run() error                // Run runs the command
 }
 
+type PlatformSpecific interface {
+	Step(image ContainerImage) (string, string, error) // Called to build each possible architecture/OS combination
+}
+
+type CrossBuilder struct {
+	PlatformSpecific
```

Comment:
This is a bit confusing... really this should be a field I think - the CrossBuilder itself does not implement `PlatformSpecific` does it?

Perhaps `builder PlatformBuilder` works and `PlatformSpecific.Step` is named `PlatformBuilder.Build` perhaps? This way CrossBuilder delegates build to a platform specific element. If I'm right that this is what's happening I think the naming would flow.